cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -846,6 +838,35 @@ private predicate nodeCandFwd2ReadTaint(Content f, Node node, boolean fromArg, C
   )
 }
 
+private predicate readCandFwd2(Content f, Configuration config) {
+  exists(Node node |
+    nodeCandFwd2Read(f, node, _, config) or
+    nodeCandFwd2ReadTaint(f, node, _, config)
+  |
+    nodeCandFwd2(node, _, _, config)
+  )
+}
+
+private predicate readStoreCandFwd2(Content f, Configuration config) {
+  readCandFwd2(f, config) and
+  storeCandFwd2(f, config)
+}
+
+private predicate summaryFwd2(Summary s, Configuration config) {
+  s = TSummaryTaint()
+  or
+  exists(Content f | s = TSummaryReadTaint(f) | readStoreCandFwd2(f, config))
+  or
+  exists(Content f | s = TSummaryTaintStore(f) | readStoreCandFwd2(f, config))
+}
+
+private predicate argumentFlowsThroughFwd2(Node n1, Node n2, Summary s, Configuration config) {
+  argumentFlowsThrough(n1, n2, _, _, s, config) and
+  nodeCandFwd2(n1, _, _, config) and
```

Comment:
no `unbind` needed?

Reply:
No, we don't need to unbind all of the `config`s, just enough that we can't get awkward joins.  Leaving this as-is just means that it can join with the first line on `config`, but then it'll also join on `n1`, which is what we want.